### PR TITLE
Use the same checkbox styles on the platform selector

### LINKED ISSUE #38997: Platforms dropdown checkbox UI is still using old look
### Issue description

Reported in p1686911660632599/1686880897.424009-slack-C01SFMVEYAK

The platforms dropdown in core profiler is still using the old UI:

Image

We should standardize to new look similar to tracking consent checkbox (particularly the extra border around checkbox and the "active" border state)

Image

### Acceptance criteria

1. Platforms dropdown UI needs to be updated to have the same look with consent checkbox.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38997 

This PR changes the look of the checkbox used on the platform selector to match the checkbox used on the first page.

### How to test the changes in this Pull Request:

1. Checkout this branch and build the assets.
2. Start the core profiler.
3. Go to `Which one of these best describes you?` page
4. Click `I'm already selling` -> `Yes, I'm selling online`
5. Click `Select platforms`
6. Check one of the checkboxes.
7. Confirm the checkbox style matches the checkbox on `Welcome to Woo!` page

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by moon0326** [comment]
Hey @verofasulo ✋ 

Could you take a look at the checkbox style changes in [update/38997-chekbox-ui-is-still-using-old-look](https://github.com/woocommerce/woocommerce/tree/update/38997-chekbox-ui-is-still-using-old-look) on a JN site please 🙏 ? 
Screen Shot 2023-07-27 at 4 22 23 PM

Screen Shot 2023-07-27 at 4 22 44 PM

**Comment by verofasulo** [comment]
> Could you take a look at the checkbox style changes in [update/38997-chekbox-ui-is-still-using-old-look](https://github.com/woocommerce/woocommerce/tree/update/38997-chekbox-ui-is-still-using-old-look) on a JN site please 🙏 ?

@moon0326 I tried, but it's not working -- not sure why. Once I click on the setup wizard, a white page is the only thing I can see.

**Comment by moon0326** [comment]
> > Could you take a look at the checkbox style changes in [update/38997-chekbox-ui-is-still-using-old-look](https://github.com/woocommerce/woocommerce/tree/update/38997-chekbox-ui-is-still-using-old-look) on a JN site please 🙏 ?
> 
> @moon0326 I tried, but it's not working -- not sure why. Once I click on the setup wizard, a white page is the only thing I can see.

I've created a JN site with the branch.

Please decode `1347395.0:722c8bff742339de57576115f65622352c5dec21021ba0696fbd6bf255ea6c5c` in once please.

**Review by chihsuan — APPROVED** [review]
I also encountered the issue of blank pages on the JN site, but I don't think it's related to this PR, as it tested well locally. LGTM! 🚢

File: plugins/woocommerce-admin/client/core-profiler/components/multiple-selector/multiple-selector.scss [hunk 1]
```diff
@@ -86,24 +92,24 @@
...
 					border-radius: 2px;
 					height: 20px !important;
 					padding: 1px;
 					width: 20px !important;
 
 					.components-checkbox-control__input[type='checkbox'] {
-						border: none;
-						height: 16px !important;
-						width: 16px !important;
+						border: 1px solid #1e1e1e;
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/core-profiler/components/multiple-selector/multiple-selector.scss]: I think the border color is a tad too strong. I think the original border color might be `#757575` Y5pUYSJPsGEud1vknUZhi8-fi-739_59745

Before:
image


After:
image

cc @verofasulo 

```suggestion
						border: 1px solid #757575;
```

**Review by ilyasfoo — APPROVED** [review]
Tested well, LGTM! Minor comment about border color, but pre-approving:

**Pull request merged**